src/ports/postgres/modules/convex/madlib_keras.py_in

```diff
@@ -0,0 +1,633 @@
+
+    plpy.info('evaluate result is {}'.format(evaluate_result))
+
+
+def evaluate1(schema_madlib, model_table, test_table, id_col, model_arch_table,
+            model_arch_id, dependent_varname, independent_varname, compile_params, output_table,
+            **kwargs):
+    # module_name = 'madlib_keras_evaluate'
+    # input_tbl_valid(test_table, module_name)
+    # input_tbl_valid(model_arch_table, module_name)
+    # output_tbl_valid(output_table, module_name)
+
+    # _validate_input_args(test_table, model_arch_table, output_table)
+
+    model_data_query = "SELECT model_data from {0}".format(model_table)
+    model_data = plpy.execute(model_data_query)[0]['model_data']
+
+    model_arch_query = "SELECT model_arch, model_weights FROM {0} " \
+                       "WHERE id = {1}".format(model_arch_table, model_arch_id)
+    query_result = plpy.execute(model_arch_query)
+    if not  query_result or len(query_result) == 0:
+        plpy.error("no model arch found in table {0} with id {1}".format(model_arch_table, model_arch_id))
+    query_result = query_result[0]
+    model_arch = query_result['model_arch']
+    input_shape = get_input_shape(model_arch)
+    compile_params = "$madlib$" + compile_params + "$madlib$"
+    # evaluate_query = plpy.prepare("""create table {output_table} as
+    evaluate_query = plpy.prepare("""
+        select {id_col}, (madlib.internal_keras_evaluate({independent_varname},
+                                             {dependent_varname},
+                                             $MAD${model_arch}$MAD$,
+                                             $1,ARRAY{input_shape},
+                                             {compile_params}))
+        from {test_table}""".format(**locals()), ["bytea"])
+    plpy.execute(evaluate_query, [model_data])
+
+def internal_keras_evaluate(x_test, y_test, model_arch, model_data, input_shape,
+                           compile_params):
+    compile_params = convert_string_of_args_to_dict(compile_params)
+    device_name = '/cpu:0'
+    os.environ["CUDA_VISIBLE_DEVICES"] = '-1'
+
+    model = model_from_json(model_arch)
+    plpy.info('model in str is {}'.format(str(model)))
+
+    model_shapes = []
+    for weight_arr in model.get_weights():
+        model_shapes.append(weight_arr.shape)
+    _, model_weights = deserialize_weights(model_data, model_shapes)
+    model.set_weights(model_weights)
+    with K.tf.device(device_name):
+        model.compile(**compile_params)
+
+    x_test = np.array(x_test).reshape(len(x_test), input_shape[0], input_shape[1],
+                                      input_shape[2])
+    x_test = x_test.astype('float32')
+    y_test = keras_utils.to_categorical(np.array(y_test), 10)
+    with K.tf.device(device_name):
+        res = model.evaluate(x_test, y_test)
+    plpy.info('evaluate result from internal_keras_evaluate is {}'.format(res))
+    return res
+
+def deserialize_iteration_state(iteration_result):
+    """
```

Comment:
These deserialization and serialization functions should be moved into `madlib_keras_helper.py_in` with the other serialization functions;  or better yet, as @kaknikhil suggests, in a separate file by themselves.